_episodes/02-benchmark-and-scaling.md

```diff
@@ -19,27 +19,244 @@ keypoints:
 
 ## What is benchmarking?
 
-(FIXME)
+Benchmarking can be viewed like a sprint athelete. The athelete runs a predetermined distance on a particular surface, and a time is recorded. Based on different conditions, such as how dry or wet the surface is, or what the surface is made of, grass, sand, or track, the times of the sprinter to cover a distance (100m, 200m, 400m etc) will differ. If you had no idea where the sprinter was running, or what the conditions were like, if the sprinter sets a certain time you can cross-correlate it with a certain time associated with certain surfaces. 
+
+Benchmarking in computing works in a similar way, as it is a way of assessing the performance of a program or set of programs. Usually codes can be tested on different computer architectures to see how a code performs. Like our sprinter, the times of benchmarks depends on a number of things, software, hardware or the computer itself and its architecture.
 
 ## What factors can affect a benchmark?
 
 (FIXME)
 
 ## Case study: Benchmarking
 
-(FIXME)
+By now you already got some understanding about how benchmarking enables you to compare the performance of your computing system with some 'standard' systems, and thus it helps you to know whether a further fine tuning of your job is required or not. 
+As a first exercise, we will start with the simplest Lennard-Jones (LJ) system as provided in the 'bench' directory of the latest LAMMPS distribution. Following this, you will be given another example and you will follow the same set of exercises and compare your results with some 'known' benchmark.
+
+> ## Run a LAMMPS job in a HPC
+> Can you list bare minimum files that you need to schedule a LAMMPS job in a HPC?
+> > ## Solution
+> > For running a LAMMPS job, we must need an input file, empirical potential file (optional), and a data file (optional). We need these optional files only when the empirical potential model parameters and the molecular coordinates are not defined within the LAMMPS input file. In addition, for submitting a LAMMPS job to a HPC queue, we need a batch file. In general, each HPC uses a queueing system to manage all the jobs. Two such popular ones are PBS and Slurm. 
+> {: .solution}
+{: .challenge}
 
-> ## Benchmarking
+The input file for the LJ-system is given below: 
+```
+{% include /snippets/ep02/in.lj %}
+```
+{: .Input}
+
+
+The timing information for this run with both 1 and 4 processors is also provided with LAMMPS distribution. So, to benchmark it would be wise to run the same job with same processor settings. Let us now create a batch file to submit this job. 
+
+
+> ## Batch file to submit a LAMMPS job
+>
+> Here is shown the header of a SLURM script to submit a LAMMPS job using 2 nodes and 48 processors. Can you modify the necessary fields in this script to submit the job using 4 processors?
```

Comment:
Another abstraction for SLURM would need to go in here depending on the system this is being run on, eg. {{ site.sched_name }} for SLURM or PBS? 

To be addressed also in config.yml